src/nspawn/nspawn.c

```diff
@@ -101,6 +104,11 @@
 #include "user-util.h"
 #include "util.h"
 
+/* Note that devpts's gid= parameter parses GIDs as signed values, hence we stay away from the upper half of the 32bit
+ * UID range here */
+#define UID_SHIFT_PICK_MIN ((uid_t) UINT32_C(0x00080000))
+#define UID_SHIFT_PICK_MAX ((uid_t) UINT32_C(0x6FFF0000))
```

Comment:
add a link to the source?
http://lxr.free-electrons.com/source/fs/devpts/inode.c?v=4.1#L189